controller.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"fmt"
+
+	compute "cloud.google.com/go/compute/apiv1"
+	"cloud.google.com/go/compute/apiv1/computepb"
+	"github.com/googleapis/gax-go/v2"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type runtimeClient interface {
+	Get(ctx context.Context, key types.NamespacedName, obj client.Object, opts ...client.GetOption) error
+}
+
+type instanceClient interface {
+	SetSecurityPolicy(ctx context.Context, req *computepb.SetSecurityPolicyInstanceRequest, opts ...gax.CallOption) (*compute.Operation, error)
+}
+
+// reconcileNode reconciles Nodes
+type reconcileNode struct {
+	// client can be used to retrieve objects from the APIServer.
+	client            runtimeClient
+	instanceClient    instanceClient
+	selector          *string
+	securityPolicyURL string
+}
+
+// Implement reconcile.Reconciler so the controller can reconcile objects
+var _ reconcile.Reconciler = &reconcileNode{}
+
+func (r *reconcileNode) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	// set up a convenient log object so we don't have to type request over and over again
+	log := log.FromContext(ctx)
+
+	// Fetch the Node from the cache
+	node := &corev1.Node{}
+	err := r.client.Get(ctx, request.NamespacedName, node)
```

Comment:
Since `Get` returns only the error, you can cuddle the error here, e.g. second example here: https://go.dev/doc/effective_go#if .. it's common practice to limit the scope.

Reply:
I've seen them a hundred times now and still forgot to use it lol